lib/buffer-search.js

```diff
@@ -224,6 +224,8 @@ class BufferSearch {
       }
 
       const newMarkers = this.createMarkers(scanStart, scanEnd);
+      if(newMarkers === false || !newMarkers.length) return; // Invalid/missing find pattern or no new markers
```

Comment:
I think what we might want to do is make the `createMarkers` return an empty array if no matches were found in the given range.